src/components/Search/SearchPageHeaderInput.tsx

```diff
@@ -0,0 +1,273 @@
+    // The input text that was last used for autocomplete; needed for the SearchRouterList when browsing list via arrow keys
+    const [autocompleteQueryValue, setAutocompleteQueryValue] = useState(textInputValue);
+
+    const [isAutocompleteListVisible, setIsAutocompleteListVisible] = useState(false);
+    const listRef = useRef<SelectionListHandle>(null);
+
+    const {type, inputQuery: originalInputQuery} = queryJSON;
+    const isCannedQuery = SearchQueryUtils.isCannedSearchQuery(queryJSON);
+    const headerText = isCannedQuery ? translate(getHeaderContent(type).titleText) : '';
+    const queryText = SearchQueryUtils.buildUserReadableQueryString(queryJSON, personalDetails, reports, taxRates);
+
+    useEffect(() => {
+        setTextInputValue(queryText);
+    }, [queryText]);
+
+    useEffect(() => {
+        const substitutionsMap = buildSubstitutionsMap(originalInputQuery, personalDetails, reports, taxRates);
+        setAutocompleteSubstitutions(substitutionsMap);
+    }, [originalInputQuery, personalDetails, reports, taxRates]);
+
+    const onSearchQueryChange = useCallback(
+        (userQuery: string) => {
+            const updatedUserQuery = SearchAutocompleteUtils.getAutocompleteQueryWithComma(textInputValue, userQuery);
+            setTextInputValue(updatedUserQuery);
+            setAutocompleteQueryValue(updatedUserQuery);
+
+            const updatedSubstitutionsMap = getUpdatedSubstitutionsMap(userQuery, autocompleteSubstitutions);
+            setAutocompleteSubstitutions(updatedSubstitutionsMap);
+
+            if (updatedUserQuery || textInputValue.length > 0) {
+                listRef.current?.updateAndScrollToFocusedIndex(0);
+            } else {
+                listRef.current?.updateAndScrollToFocusedIndex(-1);
+            }
+        },
+        [autocompleteSubstitutions, setTextInputValue, textInputValue],
+    );
+
+    const submitSearch = useCallback(
+        (queryString: SearchQueryString) => {
+            if (!queryString) {
+                return;
+            }
+
+            const cleanedQueryString = getQueryWithSubstitutions(queryString, autocompleteSubstitutions);
+            const userQueryJSON = SearchQueryUtils.buildSearchQueryJSON(cleanedQueryString);
+
+            if (!userQueryJSON) {
+                Log.alert(`${CONST.ERROR.ENSURE_BUGBOT} user query failed to parse`, {}, false);
+                return;
+            }
+
+            const standardizedQuery = SearchQueryUtils.traverseAndUpdatedQuery(userQueryJSON, SearchQueryUtils.getUpdatedAmountValue);
+            const query = SearchQueryUtils.buildSearchQueryString(standardizedQuery);
+
+            Navigation.navigate(ROUTES.SEARCH_CENTRAL_PANE.getRoute({query}));
+
+            if (query !== originalInputQuery) {
+                SearchActions.clearAllFilters();
+                setTextInputValue('');
+                setAutocompleteQueryValue('');
+                setIsAutocompleteListVisible(false);
+            }
```

Comment:
If this condition is not met we should reset the text input value to `queryText` because the text input value can be changed back and forth to the original query e.g.
1. Textinput value reflects the current search query
2. Clear textinput
3. Select previous value (from suggestion list)

Now after step 3, the `query === originalInputQuery` but the textinput is still empty.